### PR TITLE
add istioctl download instruction

### PR DESCRIPTION
**Pull request by linsun**
fixes: https://github.com/istio/istio.io/issues/5657

### CODE REVIEW

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 1]
```diff
@@ -20,6 +20,27 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+We recommend you to use a istioctl version that is the same version of your Istio control plane. Using the matching version of istioctl helps avoid unforeseen issues.
+
+### Install istioctl
+
+Install the istioctl binary with curl:
+
+1. Download the latest release with the command:
+
+{{< text bash >}}
+$ export RELEASE_VERSION="release-1.4"
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 2]
```diff
@@ -20,6 +20,27 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+$ curl -sL https://raw.githubusercontent.com/istio/istio/$RELEASE_VERSION/release/downloadIstioCtl.sh | sh -
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
$ curl -sL https://raw.githubusercontent.com/istio/istio/release-{{< istio_version >}}/release/downloadIstioCtl.sh | sh - 
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 3]
```diff
@@ -20,6 +20,27 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+Install the istioctl binary with curl:
+
+1. Download the latest release with the command:
+
+{{< text bash >}}
+$ export RELEASE_VERSION="release-1.4"
+$ curl -sL https://raw.githubusercontent.com/istio/istio/$RELEASE_VERSION/release/downloadIstioCtl.sh | sh - 
+{{< /text >}}
+
+2. Add the istioctl client to your path, on a macOS or Linux system:
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
1. Add the istioctl client to your path, on a macOS or Linux system:
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 4]
```diff
@@ -20,6 +20,27 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+
+We recommend you to use a istioctl version that is the same version of your Istio control plane. Using the matching version of istioctl helps avoid unforeseen issues.
+
+### Install istioctl
+
+Install the istioctl binary with curl:
+
+1. Download the latest release with the command:
+
+{{< text bash >}}
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]: indent 4 speces

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 5]
```diff
@@ -20,6 +20,27 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+$ export RELEASE_VERSION="release-1.4"
+$ curl -sL https://raw.githubusercontent.com/istio/istio/$RELEASE_VERSION/release/downloadIstioCtl.sh | sh - 
+{{< /text >}}
+
+2. Add the istioctl client to your path, on a macOS or Linux system:
+
+{{< text bash >}}
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]: indent 4 spaces

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 6]
```diff
@@ -20,6 +20,27 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+$ export PATH=$PATH:$HOME/.istioctl/bin 
+{{< /text >}}
+
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion

1. You can optionally enable the [auto-completion option](l#enabling-auto-completion) when working with a bash or ZSH console.
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 7]
```diff
@@ -20,6 +20,27 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
 `istioctl` only has auto-completion enabled for non-deprecated commands.
 {{< /tip >}}
 
+### Before you begin
+
+We recommend you to use a istioctl version that is the same version of your Istio control plane. Using the matching version of istioctl helps avoid unforeseen issues.
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
We recommend you to use a istioctl version that is the same version of your Istio control plane. Using the matching version of istioctl helps avoid unforeseen issues.

{{< tip >}}
If you have already [download the Istio release](/docs/setup/getting-started/#download), you should already have `istioctl` and do not need to install `istioctl` again.
{{< /tip >}}
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md (hunk 4 above)
Comment (linsun) [reply content/en/docs/ops/diagnostic-tools/istioctl/index.md]: fixed!

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md (hunk 5 above)
Comment (linsun) [reply content/en/docs/ops/diagnostic-tools/istioctl/index.md]: fixed

**Comment by linsun** [comment]
@frankbu many thanks for your review and suggestions!

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 8]
```diff
@@ -20,6 +20,33 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+
+{{< tip >}}
+If you have already [download the Istio release](/docs/setup/getting-started/#download), you should
+already have `istioctl` and do not need to install `istioctl` again.
+{{< /tip >}}
+
+### Install istioctl
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
### Install {{< istioctl >}}
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 9]
```diff
@@ -20,6 +20,33 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+
+Install the istioctl binary with curl:
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
Install the `istioctl` binary with curl:
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 10]
```diff
@@ -20,6 +20,33 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
 `istioctl` only has auto-completion enabled for non-deprecated commands.
 {{< /tip >}}
 
+### Before you begin
+
+We recommend you to use a istioctl version that is the same version of your Istio control plane. Using the matching version of istioctl helps avoid unforeseen issues.
+
+{{< tip >}}
+If you have already [download the Istio release](/docs/setup/getting-started/#download), you should
+already have `istioctl` and do not need to install `istioctl` again.
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
already have `istioctl` and do not need to install it again.
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 11]
```diff
@@ -20,6 +20,33 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+
+Install the istioctl binary with curl:
+
+1. Download the latest release with the command:
+
+    {{< text bash >}}
+    $ curl -sL https://raw.githubusercontent.com/istio/istio/release-{{< istio_version >}}/release/downloadIstioCtl.sh | sh - 
+    {{< /text >}}
+
+1. Add the istioctl client to your path, on a macOS or Linux system:
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
1. Add the `istioctl` client to your path, on a macOS or Linux system:
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 12]
```diff
@@ -20,6 +20,33 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+
+    {{< text bash >}}
+    $ export PATH=$PATH:$HOME/.istioctl/bin 
+    {{< /text >}}
+
+1. You can optionally enable the [auto-completion option](l#enabling-auto-completion) when working with a bash or ZSH console.
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
1. You can optionally enable the [auto-completion option](#enabling-auto-completion) when working with a bash or ZSH console.
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 13]
```diff
@@ -20,6 +20,33 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
 `istioctl` only has auto-completion enabled for non-deprecated commands.
 {{< /tip >}}
 
+### Before you begin
+
+We recommend you to use a istioctl version that is the same version of your Istio control plane. Using the matching version of istioctl helps avoid unforeseen issues.
+
+{{< tip >}}
+If you have already [download the Istio release](/docs/setup/getting-started/#download), you should
```
Comment (frankbu) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
If you have already [downloaded the Istio release](/docs/setup/getting-started/#download), you should
```

File: content/en/docs/ops/diagnostic-tools/istioctl/index.md [hunk 14]
```diff
@@ -20,7 +18,34 @@ The [`istioctl`](/docs/reference/commands/istioctl) tool is a configuration comm
...
+{{< /tip >}}
+
+## Install {{< istioctl >}}
+
+Install the `istioctl` binary with `curl`:
+
+1. Download the latest release with the command:
+
+    {{< text bash >}}
+    $ curl -sL https://raw.githubusercontent.com/istio/istio/release-{{< istio_version >}}/release/downloadIstioCtl.sh | sh -
```
Comment (geeknoid) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]: This is causing a problem for the linter. The linter wants you to use

```suggestion
    $ curl -sL {{< github_file >}}/release-{{< istio_version >}}/release/downloadIstioCtl.sh | sh -
```

But that ends up making a link that doesn't work for cut & paste.

I'll find a solution in a bit...

Comment (geeknoid) [reply content/en/docs/ops/diagnostic-tools/istioctl/index.md]: @linsun @howardjohn @ericvn @sdake @frankbu 

Right now, to download Istio, we tell users to use:

    $ curl -L https://git.io/getLatestIstio | ISTIO_VERSION={{< istio_full_version >}} sh -

which points to 

https://raw.githubusercontent.com/istio/istio/master/release/downloadIstioCandidate.sh

To fix the lint error, I propose replacing the two uses with:

https://istio.io/downloadIstio
https://istio.io/downloadIstoctl

These links would be automatically updated to always point to the scripts from the current release branch. We would abandon the exiting git.io URL for download.

WDYT?

Comment (linsun) [reply content/en/docs/ops/diagnostic-tools/istioctl/index.md]: That is a great idea.... I would keep the old git.io url for istio binary for backwards compatible for a release or two.

Comment (geeknoid) [reply content/en/docs/ops/diagnostic-tools/istioctl/index.md]: We can't actually remove the link from git.io, it's a read-only system. We just stop talking about it.

Comment (geeknoid) [inline content/en/docs/ops/diagnostic-tools/istioctl/index.md]:
```suggestion
    $ curl -sL https://istio.io/downloadIstioctl.sh | sh -
```

**Pull request merged**